python/ray/util/sgd/v2/trainer.py

```diff
@@ -200,7 +216,37 @@ def to_tune_trainable(
             A Trainable that can directly be passed into ``tune.run()``.
         """
 
-        def trainable_func(config: Dict[str, Any]) -> T:
-            pass
-
-        raise NotImplementedError
+        class _SgdTrainable(DistributedTrainable):
+            _function = train_func
+            # Create a new Trainer with the same configuration.
+            _trainer = Trainer(self._backend, self._num_workers, self._use_gpu,
+                               self._resources_per_worker)
+            _finished = False
+
+            @classmethod
+            def default_resource_request(cls, config: Dict) -> Resources:
+                return Resources(
+                    cpu=0,
+                    gpu=0,
+                    extra_cpu=self._num_workers,
+                    extra_gpu=self._num_workers * int(self._use_gpu))
+
+            def setup(self, config: Dict):
+                self._trainer.start_training(self._function, config)
```

Comment:
Currently in `FunctionRunner`, the training doesn't actually start until `step`. Do we want to keep that same behavior here?